tests/jballerina-unit-test/src/test/resources/test-src/statements/vardeclr/module_error_var_decl.bal

```diff
@@ -115,6 +115,27 @@ function testErrorVarDeclaredWithVar() {
     assertTrue(otherDetails3["recoverable"]);
 }
 
+type varTestErrorDetail record {
+    [int] fieldA;
+    map<boolean> fieldB;
+    error fieldC;
+};
+
+type varTestError error<varTestErrorDetail>;
+
+var error(m, error(c), fieldA = [varA], fieldB = {a: varB}, fieldC = error(msgVar)) = <varTestError>foo();
```

Comment:
Since we are testing the proper inferring of types here, shall we change `foo()`'s return type to `varTestError` instead of casting here? Applicable in other new tests too.

Reply:
Shall we also add a negative test for each scenario?

Reply:
Added